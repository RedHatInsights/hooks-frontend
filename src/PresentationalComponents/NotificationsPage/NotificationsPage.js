import React, { Fragment, Component } from 'react';
import { withRouter } from 'react-router-dom';

import {
    Main,
    PageHeader,
    PageHeaderTitle
} from '@red-hat-insights/insights-frontend-components';
import {
    Breadcrumb,
    BreadcrumbItem,
    Stack,
    StackItem
} from '@patternfly/react-core';

import PropTypes from 'prop-types';

export class NotificationsPage extends Component {
    toIndex = (event) => {
        event.preventDefault();
        this.props.history.push('/list');
    }

    showRootLink = () =>
        this.props.history && this.props.history.location.pathname !== '/list' ?
            <BreadcrumbItem to='#' onClick={ this.toIndex }>Hooks</BreadcrumbItem>
            : '';

    showBreadcrumb = () =>
        this.props.showBreadcrumb && <Breadcrumb style={ { marginLeft: 'calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)' } }>
            { this.showRootLink() }
            <BreadcrumbItem isActive style={ { marginTop: 0 } }>{ this.props.title }</BreadcrumbItem>
        </Breadcrumb>

    render() {
        const { title, children, appendix } = this.props;

        return (
            <Fragment>
                <PageHeader>
                    { this.showBreadcrumb() }
                    <PageHeaderTitle title={ title } />
                </PageHeader>
                <Stack style={ { backgroundColor: 'var(--pf-c-page__main-section--BackgroundColor)' } }>
                    <StackItem>
                        <Main style={ this.props.mainStyle }>
                            { children }
                        </Main>
                    </StackItem>
                    { appendix &&
                      <StackItem>
                          { appendix }
                      </StackItem> }
                </Stack>
            </Fragment>
        );
    }
};

NotificationsPage.defaultProps = {
    showBreadcrumb: true
};

NotificationsPage.propTypes = {
    title: PropTypes.string.isRequired,
    rightBar: PropTypes.node,
    showBreadcrumb: PropTypes.bool,
    children: PropTypes.node,
    history: PropTypes.object,
    mainStyle: PropTypes.object,
    appendix: PropTypes.node
};

export default withRouter(NotificationsPage);
