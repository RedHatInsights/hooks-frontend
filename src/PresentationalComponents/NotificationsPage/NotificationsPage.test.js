import { NotificationsPage } from './NotificationsPage';

describe('NotificationsPage', () => {
    it('expect to render ', () => {
        const wrapper = shallow(
            <NotificationsPage title="Something" appendix={ <strong>A footer</strong> }>
                content
            </NotificationsPage>
        );
        expect(wrapper).toMatchSnapshot();
    });
});
