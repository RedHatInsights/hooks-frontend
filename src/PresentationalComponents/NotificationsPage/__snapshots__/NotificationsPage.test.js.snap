// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsPage expect to render  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotificationsPage
    appendix={
      <strong>
        A footer
      </strong>
    }
    showBreadcrumb={true}
    title="Something"
  >
    content
  </NotificationsPage>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <PageHeader>
          <Breadcrumb
            aria-label="Breadcrumb"
            className=""
            style={
              Object {
                "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
              }
            }
          >
            
            <BreadcrumbItem
              className=""
              component="a"
              isActive={true}
              style={
                Object {
                  "marginTop": 0,
                }
              }
              target={null}
              to={null}
            >
              Something
            </BreadcrumbItem>
          </Breadcrumb>
          <PageHeaderTitle
            title="Something"
          />
        </PageHeader>,
        <Stack
          className=""
          component="div"
          gutter={null}
        >
          <StackItem
            className=""
            isFilled={false}
          >
            <Connect(Main)>
              content
            </Connect(Main)>
          </StackItem>
          <StackItem
            className=""
            isFilled={false}
          >
            <strong>
              A footer
            </strong>
          </StackItem>
        </Stack>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Breadcrumb
              aria-label="Breadcrumb"
              className=""
              style={
                Object {
                  "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
                }
              }
            >
              
              <BreadcrumbItem
                className=""
                component="a"
                isActive={true}
                style={
                  Object {
                    "marginTop": 0,
                  }
                }
                target={null}
                to={null}
              >
                Something
              </BreadcrumbItem>
            </Breadcrumb>,
            <PageHeaderTitle
              title="Something"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "aria-label": "Breadcrumb",
              "children": Array [
                "",
                <BreadcrumbItem
                  className=""
                  component="a"
                  isActive={true}
                  style={
                    Object {
                      "marginTop": 0,
                    }
                  }
                  target={null}
                  to={null}
                >
                  Something
                </BreadcrumbItem>,
              ],
              "className": "",
              "style": Object {
                "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
              },
            },
            "ref": null,
            "rendered": Array [
              "",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Something",
                  "className": "",
                  "component": "a",
                  "isActive": true,
                  "style": Object {
                    "marginTop": 0,
                  },
                  "target": null,
                  "to": null,
                },
                "ref": null,
                "rendered": "Something",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "title": "Something",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <StackItem
              className=""
              isFilled={false}
            >
              <Connect(Main)>
                content
              </Connect(Main)>
            </StackItem>,
            <StackItem
              className=""
              isFilled={false}
            >
              <strong>
                A footer
              </strong>
            </StackItem>,
          ],
          "className": "",
          "component": "div",
          "gutter": null,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Connect(Main)>
                content
              </Connect(Main)>,
              "className": "",
              "isFilled": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "content",
                "style": undefined,
              },
              "ref": null,
              "rendered": "content",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <strong>
                A footer
              </strong>,
              "className": "",
              "isFilled": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A footer",
              },
              "ref": null,
              "rendered": "A footer",
              "type": "strong",
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <PageHeader>
            <Breadcrumb
              aria-label="Breadcrumb"
              className=""
              style={
                Object {
                  "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
                }
              }
            >
              
              <BreadcrumbItem
                className=""
                component="a"
                isActive={true}
                style={
                  Object {
                    "marginTop": 0,
                  }
                }
                target={null}
                to={null}
              >
                Something
              </BreadcrumbItem>
            </Breadcrumb>
            <PageHeaderTitle
              title="Something"
            />
          </PageHeader>,
          <Stack
            className=""
            component="div"
            gutter={null}
          >
            <StackItem
              className=""
              isFilled={false}
            >
              <Connect(Main)>
                content
              </Connect(Main)>
            </StackItem>
            <StackItem
              className=""
              isFilled={false}
            >
              <strong>
                A footer
              </strong>
            </StackItem>
          </Stack>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Breadcrumb
                aria-label="Breadcrumb"
                className=""
                style={
                  Object {
                    "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
                  }
                }
              >
                
                <BreadcrumbItem
                  className=""
                  component="a"
                  isActive={true}
                  style={
                    Object {
                      "marginTop": 0,
                    }
                  }
                  target={null}
                  to={null}
                >
                  Something
                </BreadcrumbItem>
              </Breadcrumb>,
              <PageHeaderTitle
                title="Something"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "aria-label": "Breadcrumb",
                "children": Array [
                  "",
                  <BreadcrumbItem
                    className=""
                    component="a"
                    isActive={true}
                    style={
                      Object {
                        "marginTop": 0,
                      }
                    }
                    target={null}
                    to={null}
                  >
                    Something
                  </BreadcrumbItem>,
                ],
                "className": "",
                "style": Object {
                  "marginLeft": "calc(var(--pf-c-content--ol--MarginLeft) * -1 * 2)",
                },
              },
              "ref": null,
              "rendered": Array [
                "",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Something",
                    "className": "",
                    "component": "a",
                    "isActive": true,
                    "style": Object {
                      "marginTop": 0,
                    },
                    "target": null,
                    "to": null,
                  },
                  "ref": null,
                  "rendered": "Something",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "title": "Something",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <StackItem
                className=""
                isFilled={false}
              >
                <Connect(Main)>
                  content
                </Connect(Main)>
              </StackItem>,
              <StackItem
                className=""
                isFilled={false}
              >
                <strong>
                  A footer
                </strong>
              </StackItem>,
            ],
            "className": "",
            "component": "div",
            "gutter": null,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Connect(Main)>
                  content
                </Connect(Main)>,
                "className": "",
                "isFilled": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "content",
                  "style": undefined,
                },
                "ref": null,
                "rendered": "content",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <strong>
                  A footer
                </strong>,
                "className": "",
                "isFilled": false,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A footer",
                },
                "ref": null,
                "rendered": "A footer",
                "type": "strong",
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
